internal/provider/resource_ably_namespace_test.go

```diff
@@ -0,0 +1,83 @@
+package ably_control
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"testing"
+)
+
+func TestAccAblyNamespace(t *testing.T) {
+	app_name := acctest.RandStringFromCharSet(15, acctest.CharSetAlphaNum)
+	namespace_name := acctest.RandStringFromCharSet(15, acctest.CharSetAlphaNum)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			// Create and Read testing of ably_app.app0
+			{
+				Config: testAccAblyNamespaceConfig(app_name, namespace_name, true, true, true, true, true, true),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ably_app.app0", "name", app_name),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "id", namespace_name),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "authenticated", "true"),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "persisted", "true"),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "persist_last", "true"),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "push_enabled", "true"),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "tls_only", "true"),
+					resource.TestCheckResourceAttr("ably_namespace.namespace0", "expose_timeserial", "true"),
+				),
+			},
+			// Update and Read testing of ably_app.app0
+			{
+				Config: testAccAblyNamespaceConfig(app_name, namespace_name, false, false, false, false, false, false),
```

Comment:
As part of the test, change the namespace name as well to ensure that namespace name update works correctly.

Reply:
The name can't be changed.